pkg/controller/windowsmachineconfig/nodeconfig/nodeconfig.go

```diff
@@ -48,6 +51,53 @@ func (nc *nodeConfig) Configure() error {
 	if err := nc.handleCSRs(); err != nil {
 		return errors.Wrap(err, "handling CSR for the given node failed")
 	}
+
+	var instanceID string
+	// As the CSR has been approved get the Kubernetes node object associated
+	// the Windows VM created
+	if nc.Windows.GetCredentials() != nil && len(nc.Windows.GetCredentials().GetInstanceId()) > 0 {
+		instanceID = nc.Windows.GetCredentials().GetInstanceId()
+	}
+
+	// Get the node object associated with node
+	node, err := nc.getNodeObjectAssociated(instanceID)
+	if err != nil {
+		return errors.Wrap(err, "node object associated with the node")
+	}
+
+	// Apply worker labels
+	if err := nc.applyWorkerLabel(node); err != nil {
+		return errors.Wrap(err, "failed applying worker label")
+	}
+	return nil
+}
+
+func (nc *nodeConfig) getNodeObjectAssociated(instanceID string) (*v1.Node, error) {
+	nodes, err := nc.k8sclientset.CoreV1().Nodes().List(metav1.ListOptions{LabelSelector: "node.openshift.io/os_id=Windows"})
+	if err != nil {
+		return nil, errors.Wrap(err, "error listing nodes")
+	}
+	for _, node := range nodes.Items {
+		if instanceID == GetInstanceID(node.Spec.ProviderID) {
+			return &node, nil
+		}
+	}
+	return nil, errors.Errorf("unable to find node for instanceID %s", instanceID)
+}
+
+// GetInstanceID gets the instanceID of VM for a given cloud provider ID
+// Ex: aws:///us-east-1e/i-078285fdadccb2eaa. We always want the last entry which is the instanceID
+func GetInstanceID(providerID string) string {
```

Comment:
You already have this method in test suite. I made it public to be shared with the test suite, is that why you made it public while copying too?

Reply:
should i remove the one in test suite and use this one instead ?

Reply:
You should, if you want to re-use it there unless @aravindhp has any concerns.

Reply:
I prefer not sharing test and main code.